build/publish.proj

```diff
@@ -25,43 +30,33 @@
 
     <Error Condition="'@(ItemsToPush)' == ''" Text="No packages to push." />
 
-    <ItemGroup>
-      <BuildData Include="Location=$(FeedUrl)" />
-    </ItemGroup>
-
-    <Error Condition="'$(AccountKey)' == ''" Text="The AccountKey property is required." />
-    <Error Condition="'$(BuildId)' == ''" Text="The BuildId property is required." />
-    <Error Condition="'$(ManifestFilePath)' == ''" Text="The ManifestFilePath property is required." />
-    <Error Condition="'$(RepoBranch)' == ''" Text="The RepoBranch property is required." />
-    <Error Condition="'$(RepoCommit)' == ''" Text="The RepoCommit property is required." />
-    <Error Condition="'$(RepoUri)' == ''" Text="The RepoUri property is required." />
+    <Error Condition="'$(BUILD_BUILDNUMBER)' == ''" Text="The BUILD_BUILDNUMBER property is required." />
+    <Error Condition="'$(ArtifactsLogDir)' == ''" Text="The ArtifactsLogDir property is required." />
+    <Error Condition="'$(BUILD_SOURCEBRANCH)' == ''" Text="The BUILD_SOURCEBRANCH property is required." />
+    <Error Condition="'$(BUILD_SOURCEVERSION)' == ''" Text="The BUILD_SOURCEVERSION property is required." />
+    <Error Condition="'$(BUILD_REPOSITORY_URI)' == ''" Text="The BUILD_REPOSITORY_URI property is required." />
+    <Error Condition="'$(MaestroAccessToken)' == ''" Text="The MaestroAccessToken property is required." />
+    <Error Condition="'$(MaestroApiEndpoint)' == ''" Text="The MaestroApiEndpoint property is required." />
 
     <Message Text="Publishing %(ItemsToPush.Identity)" Importance="normal" />
 
-    <PushToBlobFeed
-      AccountKey="$(AccountKey)"
-      AssetManifestPath="$(ManifestFilePath)"
-      ExpectedFeedUrl="$(FeedUrl)"
+    <PushToAzureDevOpsArtifacts
       ItemsToPush="@(ItemsToPush)"
-      ManifestBranch="$(RepoBranch)"
-      ManifestBuildId="$(BuildId)"
-      ManifestBuildData="$(BuildData)"
-      ManifestCommit="$(RepoCommit)"
-      ManifestRepoUri="$(RepoUri)"
-      MaxClients="8"
-      Overwrite="false"
-      PassIfExistingItemIdentical="true" />
-  </Target>
+      ManifestBuildData="Location=$(FeedUrl)"
```

Comment:
Take a look how we do this in the Arcade.SDK, I think it'll help you simplify things: https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj#L123 also, it'll be better if you set `InitialAssetsLocation=....` instead of just `Location=...`